include/oneapi/dpl/pstl/iterator_impl.h

```diff
@@ -75,6 +76,38 @@ struct __make_references
     }
 };
 
+template <typename _Iter, typename = void>
+struct __is_legacy_passed_directly : std::false_type
+{
+};
+
+template <typename _Iter>
+struct __is_legacy_passed_directly<_Iter, ::std::enable_if_t<_Iter::is_passed_directly::value>> : std::true_type
+{
+};
+
+template <typename _T>
+struct __is_reversed_indirectly_device_accessible_it;
+
+template <typename T>
+constexpr auto is_onedpl_indirectly_device_accessible_iterator(T)
```

Comment:
So too long name...
BTW, due to "indirectly" is confusing term... may be we omit that?)